core/src/main/java/org/mapfish/print/map/image/wms/WmsLayer.java

```diff
@@ -69,10 +78,16 @@ protected BufferedImage loadImage(final MfClientHttpRequestFactory requestFactor
             final BufferedImage image = ImageIO.read(response.getBody());
             if (image == null) {
                 LOGGER.warn("The URI: " + uri + " is an image format that can be decoded");
+                this.registry.counter(baseMetricName + ".error").inc();
                 return createErrorImage(paintArea);
+            } else {
+                timerDownload.stop();
             }
 
             return image;
+        } catch (Throwable e) {
+            this.registry.counter(baseMetricName + ".error").inc();
+            throw e;
         } finally {
             closer.close();
```

Comment:
No, I want it done only in case of success... in case of error we just increment the error counter.

If stop is not done, no resources is leaked. Don't worry.